Sodium/Sodium.swift

```diff
@@ -9,6 +9,8 @@ public struct Sodium {
     public let randomBytes = RandomBytes()
     public let shortHash = ShortHash()
     public let sign = Sign()
+    public let sha256Hash = SHA256Hash()
+    public let sha512Hash = SHA512Hash()
```

Comment:
I'm wondering about the API. Seems like this deviates from the general pattern of swift-sodium compared to libsodium. How about adding a `cryptoHash` layer that encapsulates the SHA-256/512 implementations. Feels a bit odd to have very specific algorithms exposed at the top, given the main idea behind libsodium to offer high-level interfaces.

Reply:
Ping @blochberger . This feels close to landing and I'd like your help to get it there.

Reply:
Re-ping @blochberger

Reply:
I agree that having these functions at the top-level feels very odd. They are low-level primitives, and not high-level functions.

Ideally, there should be a new namespace for primitives, with another namespace for hash functions inside.

Also, SHA512-256 is missing even though this is the safest SHA2-based option.

Reply:
> > I can't find this functionality in the C code. There's `crypto_auth_hmacsha512256` but that's hmac.
> Apparently a SHA512-256 is not just a truncated SHA512 so I won't risk an implementation if it's not exposed.

My bad, I thought `crypto_hash()` was truncated SHA512 like `crypto_auth()` so it's not vulnerable to length extensions attacks, but it's indeed just SHA512.

Reply:
All right. Could you also confirm if the commit I linked is the changes you need to merge this PR, and otherwise provide more info or push a commit doing it?

Reply:
I was rather thinking of something like `primitives.hash.sha256` .